### PR TITLE
Xenial migration

### PR DESCRIPTION
**Pull request by ipa-nhg**
Update the scripts to support xenial, copied parts of the code from the ros-buildfarm jobs.

### CODE REVIEW

**Comment by pbeeson** [comment]
Is this working with the current kinetic rosdistro ?

File: buildbot_ros_cfg/distro.py [hunk 1]
```diff
@@ -210,9 +210,13 @@ def debbuilders_from_rosdistro(c, oracle, distro, builders):
                 for code_name in build_file.get_target_os_code_names(os):
                     for arch in build_file.get_target_arches(os, code_name):
                         print('Configuring ros_debbuild job for: %s_%s_%s' % (name, code_name, arch))
+			try:
```
Comment (mikeferguson) [inline buildbot_ros_cfg/distro.py]: this is using Tabs -- please replace with spaces.

File: buildbot_ros_cfg/ros_deb.py [hunk 2]
```diff
@@ -108,7 +108,7 @@ def ros_debbuild(c, job_name, packages, url, distro, arch, rosdistro, version, m
             ShellCommand(
                 haltOnFailure = True,
                 name = package+'-stampdeb',
-                command = ['git-dch', '-a', '--ignore-branch', '--verbose',
+                command = ['gbp', 'dch', '-a', '--ignore-branch', '--verbose',
```
Comment (mikeferguson) [inline buildbot_ros_cfg/ros_deb.py]: does this work with older distros as well. or do we need to fork for xenial? I don't have access to a test machine at the moment to check.

File: scripts/aptrepo-create.bash [hunk 3]
```diff
@@ -17,7 +17,7 @@ else
     export ARCHS=${2}
 fi
 if [[ ${#} -lt 3 ]]; then
-    export DISTROS=(trusty)
+    export DISTROS=(xenial)
```
Comment (mikeferguson) [inline scripts/aptrepo-create.bash]: I would recommend we remove the defaults entirely and just update the README to have people put in the correct distros. Alternatively, we should add a message here so we don't get a silent parameter substitution that people aren't aware of.

File: scripts/build_source_deb.py [hunk 4]
```diff
@@ -20,33 +20,45 @@
...
+	# dpkg-buildpackage args
+	'-us', '-uc']
+	# debuild args for lintian
+	#'--lintian-opts', '--suppress-tags', 'newer-standards-version']
+
+cmd += ['--git-upstream-tree=TAG',
         '--git-upstream-tag=release/{rosdistro}/{package}/{release_version}'.format(
             rosdistro=rosdistro, package=package, release_version=release_version)] + gbp_args
-else:
-    command = ['git-buildpackage', '-S'] + gbp_args
```
Comment (mikeferguson) [inline scripts/build_source_deb.py]: it seems this PR drops support for older versions of git-buildpackage -- but I think that only affects 12.04 (based off a quick review of https://github.com/mikeferguson/buildbot-ros/pull/39 which introduced these files). Is this expected/tested to still work on 14.04?

File: scripts/build_binary_deb.py [hunk 5]
```diff
@@ -19,37 +19,66 @@
...
-else:
-    command = ['git-buildpackage', '--git-pbuilder', '--git-export=WC',
-      '--git-export-dir={workdir}'.format(workdir=workdir)] + gbp_args
-
-# Call out to git-buildpackage
-print 'Running git-buildpackage command: ' + str(command)
-sys.stdout.flush()
-os.execlp('git-buildpackage', *command)
+try:
+	subfolders = _get_package_subfolders(workdir, debian_pkg)
```
Comment (mikeferguson) [inline scripts/build_binary_deb.py]: tabs here too

File: scripts/build_source_deb.py [hunk 6]
```diff
@@ -20,33 +20,45 @@
...
+            continue
+        if filename.startswith('%s-' % debian_package_name):
+            subfolders.append(path)
+    return subfolders
+
+rosdistro, package, release_version, workdir = sys.argv[1:5]
+gbp_args = sys.argv[5:]
+
+try:
+	subfolders = _get_package_subfolders(workdir, debian_pkg)
```
Comment (mikeferguson) [inline scripts/build_source_deb.py]: and tabs here

**Review by mikeferguson — CHANGES_REQUESTED** [review]
Biggest issue is the tabs in several files. Secondary is to define whether we need to also tag/branch if this no longer supports trusty in the build_source_deb.py and build_binary_deb.py scripts.

File: buildbot_ros_cfg/ros_deb.py (hunk 2 above)
Comment (ipa-nhg) [reply buildbot_ros_cfg/ros_deb.py]: gbp is also supported for trusty http://manpages.ubuntu.com/manpages/xenial/man1/gbp-buildpackage.1.html , I tried the command line and it is working for 14.04

File: scripts/build_source_deb.py (hunk 4 above)
Comment (ipa-nhg) [reply scripts/build_source_deb.py]: This configuration was successfully tested on a xenial machine and for kinetic repositories. Theoretically should be ok for trusty, I have a second buildbot system running on a 14.04 server, I hope I have time  this week to test this version and confirm that it works.

I showed you had before 2 command lines depending on the buildpackage version to support 12.04, alternatively we could have a get_linux_distro function and call "git-buildpackage" or "gpb buildpackage" depending on the value. Unfortunately I will not be able to test a 12.04 configuration.

**Comment by AravindaDP** [comment]
Any updates on this?

File: scripts/build_source_deb.py [hunk 7]
```diff
@@ -20,33 +20,45 @@
...
+    subfolders = []
+    for filename in os.listdir(basepath):
+        path = os.path.join(basepath, filename)
+        if not os.path.isdir(path):
+            continue
+        if filename.startswith('%s-' % debian_package_name):
+            subfolders.append(path)
+    return subfolders
+
+rosdistro, package, release_version, workdir = sys.argv[1:5]
```
Comment (AravindaDP) [inline scripts/build_source_deb.py]: By the way I get following error:

/home/buildbot/rosbuilder1/arbotix_indigo_trusty_amd64_debbuild/build_source_deb.py indigo arbotix_msgs 0.10.0-0 -uc -us --git-ignore-branch --git-ignore-new --git-verbose --git-dist=trusty --git-arch=amd64
 in dir /home/buildbot/rosbuilder1/arbotix_indigo_trusty_amd64_debbuild/build (timeout 1200 secs)
 watching logfiles {}
 argv: ['/home/buildbot/rosbuilder1/arbotix_indigo_trusty_amd64_debbuild/build_source_deb.py', 'indigo', 'arbotix_msgs', '0.10.0-0', '-uc', '-us', '--git-ignore-branch', '--git-ignore-new', '--git-verbose', '--git-dist=trusty', '--git-arch=amd64']
 environment:
  HOME=/home/buildbot
  HUSHLOGIN=FALSE
  LANG=en_US.UTF-8
  LESSCLOSE=/usr/bin/lesspipe %s %s
  LESSOPEN=| /usr/bin/lesspipe %s
  LOGNAME=buildbot
  LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
  MAIL=/var/mail/buildbot
  PATH=/home/buildbot/buildbot-ros/scripts:/home/buildbot/buildbot-env/bin:/home/buildbot/bin:/home/buildbot/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
  PS1=(buildbot-env) ${debian_chroot:+($debian_chroot)}\u@\h:\w\$ 
  PWD=/home/buildbot/rosbuilder1/arbotix_indigo_trusty_amd64_debbuild/build
  SHELL=/bin/bash
  SHLVL=1
  TERM=linux
  USER=buildbot
  VIRTUAL_ENV=/home/buildbot/buildbot-env
  XDG_RUNTIME_DIR=/run/user/1000
  XDG_SEAT=seat0
  XDG_SESSION_ID=1
  XDG_VTNR=1
  _=/home/buildbot/buildbot-env/bin/buildslave
 using PTY: False
Invoking 'gbp buildpackage --git-ignore-new --git-ignore-branch -S -us -uc --git-upstream-tree=TAG --git-upstream-tag=release/indigo/arbotix_msgs/0.10.0-0 -us --git-ignore-branch --git-ignore-new --git-verbose --git-dist=trusty --git-arch=amd64' in '-uc/build'
Traceback (most recent call last):
  File "/home/buildbot/rosbuilder1/arbotix_indigo_trusty_amd64_debbuild/build_source_deb.py", line 58, in <module>
    subprocess.check_call(cmd,cwd=sources_dir,env=env)
  File "/usr/lib/python2.7/subprocess.py", line 536, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib/python2.7/subprocess.py", line 523, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib/python2.7/subprocess.py", line 711, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1343, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory: '-uc/build'
program finished with exit code 1

**Comment by mikeferguson** [comment]
No longer supported, about to archive

**Pull request closed without merging**